tls/s2n_resume.c

```diff
@@ -1027,16 +1026,26 @@ int s2n_config_wipe_expired_ticket_crypto_keys(struct s2n_config *config, int8_t
 
 end:
     for (int j = 0; j < num_of_expired_keys; j++) {
-        POSIX_GUARD_RESULT(s2n_set_remove(config->ticket_keys, expired_keys_index[j] - j));
+        POSIX_GUARD_RESULT(s2n_array_remove(config->ticket_keys, expired_keys_index[j] - j));
     }
 
     return 0;
 }
 
 int s2n_config_store_ticket_key(struct s2n_config *config, struct s2n_ticket_key *key)
 {
-    /* Keys are stored from oldest to newest */
-    POSIX_GUARD_RESULT(s2n_set_add(config->ticket_keys, key));
+    uint32_t ticket_keys_len = 0;
+    POSIX_GUARD_RESULT(s2n_array_num_elements(config->ticket_keys, &ticket_keys_len));
+
+    /* The ticket key secret must be unique. */
+    for (uint32_t i = 0; i < ticket_keys_len; i++) {
```

Comment:
Probably also not a blocker, but I realized we must also be checking for id uniqueness somewhere in the process, since ids must also be unique. We are: https://github.com/aws/s2n-tls/blob/ac1d09851b4d625f83d1fa68b467abbca8fdd408/tls/s2n_config.c#L969-L972

It would probably make more sense (and save us an iteration) if we did both things at the same time.